autogen/main/variables.tf.tmpl

```diff
@@ -114,6 +114,23 @@ variable "insecure_kubelet_readonly_port_enabled" {
   default     = null
 }
 
+{% if autopilot_cluster %}
+variable "node_pools_cgroup_mode" {
+  type = string
+  description = "Specifies the Linux cgroup mode for autopilot Kubernetes nodes in the cluster. Accepted values are `CGROUP_MODE_UNSPECIFIED`, `CGROUP_MODE_V1`, and `CGROUP_MODE_V2`, which determine the control group hierarchy used for resource management."
+  validation {
+    condition = var.node_pools_cgroup_mode == null || contains([
```

Comment:
Unfortunately Terraform doesn't short circuit evaluation, so you will need a `try()` around `contains()` for `null` handling.

Reply:
10 pts for Gryffindor, I'm wrong on coalesce behaviour, I revert my `coalesce` commit

Reply:
I use this line to avoid NPE `var.node_pools_cgroup_mode != null ? var.node_pools_cgroup_mode : ""`